stub-idp/src/test/java/uk/gov/ida/stub/idp/repositories/jdbc/JDBIIdpSessionRepositoryTest.java

```diff
@@ -33,7 +34,33 @@ public void setUp() {
 		jdbi = Jdbi.create(DATABASE_URL);
 		repository = new JDBIIdpSessionRepository(jdbi);
 	}
-	
+
+	@Test
+	public void testSessionDeletion() throws InterruptedException {
+		// force clear all sessions from the test database
+		repository.deleteSessionsOlderThan(Duration.ZERO);
+		// this is not ideal - and could be flaky.
+		// DateTimeFreezer could be updated and used but that has proved difficult...
+		final long sleepSeconds = 5;
+
+		assertThat(repository.countSessionsInDatabase()).isZero().withFailMessage("found %d sessions in the database when it should be empty", repository.countSessionsInDatabase());
+		repository.createSession(createSession());
+		assertThat(repository.countSessionsInDatabase()).isEqualTo(1);
+		Thread.sleep(sleepSeconds*1000);
```

Comment:
anyway we could do this without a sleep?

Reply:
I tried - see the latest commit message for the test - it needs a `Clock` to be used in `JDBIIdpSessionRepository` but I could not get it to work

Reply:
fair enough, if it becomes too annoying then I'm sure somebody else could try and fix it later.